helm/cluster-eks/templates/_awsaccount.tpl

```diff
@@ -0,0 +1,22 @@
+{{- /*
+Extracts the AWS account ID from an ARN string.
+
+Example usage: {{ include "extractAWSAccountID" "arn:aws:iam::1234567890:role/example-role" }}
+
+Input: An ARN string
+Output: The AWS account ID
+*/ -}}
+{{- define "extractAWSAccountID" -}}
+{{- $parts := (split ":" .)  -}}
+{{- if ge (len $parts) 5 -}}{{- $parts._4 -}}{{- end -}}
+{{- end -}}
+
+{{- define "aws-account-id" -}}
+{{- $roleName:= .Values.providerSpecific.awsClusterRoleIdentityName -}}
+{{- $accountID := "*" -}}
```

Comment:
If the `AWSClusterRoleIdentity` isn't found, is it okay for us to return `"*"`? Wouldn't it be better to error out?

Reply:
that's probably good idea to error here

default `AWSClusterRoleIdentity` is created by `mc-bootstrap` and any additional is manual as its depend if the customer wants to deploy to multiple different accounts or not

Reply:
damn in the `template` and `lint` version of the helm command the `lookup` function always return empty fields as it does no uses the k8s, so there is no way I can make this pass the test :( not sure how to address this

Reply:
I don't know how to workaround it either. Maybe we need to expose a new value `awsAccountId`, and the lookup is only used when `awsAccountId` is empty. For these tests, we pass an `awsAccountId`.

Reply:
fixed